ports/psoc6/mpconfigport.h

```diff
@@ -86,139 +84,66 @@
-#endif
-
-#ifndef MICROPY_PY_USSL
-#define MICROPY_PY_USSL                 (1)
-#endif
-
-#ifndef MICROPY_PY_UWEBSOCKET
-#define MICROPY_PY_UWEBSOCKET           (1)
-#endif
-
-#ifndef MICROPY_PY_UHASHLIB_SHA1
-#define MICROPY_PY_UHASHLIB_SHA1        (1)
-#endif
+// #define MICROPY_PY_NETWORK                      (1)
+// #define MICROPY_PY_USOCKET                      (1)
+// #define MICROPY_PY_UWEBSOCKET                   (1)
+// #define MICROPY_PY_WEBREPL                      (1)
+#define MICROPY_PY_OS_DUPTERM                      (1)
 
-#ifndef MICROPY_PY_WEBREPL
-#define MICROPY_PY_WEBREPL              (1)
-#endif
+#define MP_STATE_PORT                           MP_STATE_VM
 
-#ifndef MICROPY_PY_OS_DUPTERM
-#define MICROPY_PY_OS_DUPTERM           (1)
-#endif
+#define MICROPY_PY_UCRYPTOLIB                   (1)
+#define MICROPY_PY_UCRYPTOLIB_CTR               (1)
+#define MICROPY_PY_UCRYPTOLIB_CONSTS            (1)
 
-#endif
-
-
-#define MICROPY_TRACKED_ALLOC        (MICROPY_SSL_MBEDTLS)
-
-#define MICROPY_PY_UCRYPTOLIB        (1)
-#define MICROPY_PY_UCRYPTOLIB_CTR    (1)
-#define MICROPY_PY_UCRYPTOLIB_CONSTS (1)
-
-#define MICROPY_PY_UHASHLIB          (1)
-#define MICROPY_PY_UHASHLIB_MD5      (1)
-#define MICROPY_PY_UHASHLIB_SHA1     (1)
-#define MICROPY_PY_UHASHLIB_SHA256   (1)
-
-#define MP_STATE_PORT MP_STATE_VM
+#define MICROPY_PY_UHASHLIB_MD5                 (1)
+#define MICROPY_PY_UHASHLIB_SHA1                (1)
 
 // Miscellaneous settings
-#define MICROPY_MAKE_POINTER_CALLABLE(p) ((void *)((mp_uint_t)(p) | 1))
+#define MICROPY_MAKE_POINTER_CALLABLE(p)        ((void *)((mp_uint_t)(p) | 1))
 
 #define MP_SSIZE_MAX (0x7fffffff)
-typedef intptr_t mp_int_t; // must be pointer size
+typedef intptr_t mp_int_t;  // must be pointer size
 typedef uintptr_t mp_uint_t; // must be pointer size
 typedef intptr_t mp_off_t;
 
 // We need to provide a declaration/definition of alloca()
 #include <alloca.h>
 
+#define MICROPY_PY_BUILTINS_HELP_TEXT           psoc6_help_text
 
-////////////////////////////////////////////////////////////////////////
-// TODO: Remove options implicitly set by setting feature level. Must work for all feature levels !
-// the following entries have been added by JB
-// Usually passed from Makefile
-#ifndef MICROPY_GC_HEAP_SIZE
-#define MICROPY_GC_HEAP_SIZE (256 * 1024)
-#endif
-
-
-#define MICROPY_LOGGER_DEBUG              (0)
-
-
-#define MICROPY_PY_BUILTINS_HELP          (1)
-#define MICROPY_PY_BUILTINS_HELP_MODULES  (1)
-
-
-
-#define MICROPY_ENABLE_COMPILER           (1)
-#define MICROPY_QSTR_EXTRA_POOL           mp_qstr_frozen_const_pool
-#define MICROPY_ALLOC_PARSE_CHUNK_INIT    (16)
-#define MICROPY_KBD_EXCEPTION             (1)
-
-#define MICROPY_ENABLE_SOURCE_LINE        (1)
 
-#define MICROPY_USE_INTERNAL_PRINTF       (0)
-#define MICROPY_ENABLE_FINALISER          (1)
-#define MICROPY_VFS_POSIX                 (0)
-#define MICROPY_READER_POSIX              (0)
+// set in Makefile
```

Comment:
Could we remove if already set in Makefile?

Reply:
done.